File: backend/src/LibraryBackend/DTOs/Requests/SaveBookRequest.cs
namespace LibraryBackend.DTOs.Requests;

public class SaveBookRequest
{
    public int? Id { get; init; }

    public string Title { get; init; } = null!;

    /// <summary>
    /// Base64 encoded image.
    /// Can be null when <see cref="Id"/> is not null.
    /// </summary>
    public string? Cover { get; init; } = null!;

    public string Content { get; init; } = null!;

    public string Genre { get; init; } = null!;

    public string Author { get; init; } = null!;
}

File: frontend/src/modules/book/dto/requests/save-book-request.ts
export class SaveBookRequest {
  id: number | null;
  title: string;
  cover: string | null;
  content: string;
  genre: string;
  author: string;

  constructor(
    id: number | null,
    title: string,
    cover: string | null,
    content: string,
    genre: string,
    author: string) {

    this.id = id;
    this.title = title;
    this.cover = cover;
    this.content = content;
    this.genre = genre;
    this.author = author;
  }
}
